doc/Security.xml

```diff
@@ -947,6 +979,55 @@ Added certificate-based client authentication</revremark>
         </orderedlist>
       </section>
     </section>
+    <section xml:id="_Ref395181339">
+      <title>JWT-based client authorization</title>
+      <para>Unlike with HTTP Digest and WS-UsernameToken authentication, this specification defines how to associate clients with the different User Levels as defined in the ONVIF Core Specification. Devices are not expected to have the user credentials stored in their memory, instead they will verify that the user has been correctly authenticated by an external service based on Open ID Connect and authorize accessing different functions, based on the user level presented in  the supplied JWT embedded in the binary security token.</para>
+      <para>The content of the binary security token is the base64-encoded JWT according to [RFC 7519].</para>
+      <para>The JOSE header shall declare that the encoded object is a JWT, and the JWT is a JWS that is signed using ECDSA using P-256 (secp256r1) curve and SHA-256 hash algorithm. Additionally, it must include the <emphasis>kid</emphasis> claim as defined by [RFC 7515], which indicates the ID associated to the key used to sign the JWT. If the absolute path of the JWK Set URL is different from https://iss/.well-known/jwks.json, the header may include an optional <emphasis>jku</emphasis> claim.</para>
+      <para>The JWT shall include the following standard claims:</para>
+      <para>
+        <itemizedlist>
+          <listitem>
+            <para><emphasis>iss</emphasis>: Issuer, the URI of the server that generated the JWT.</para>
+          </listitem>
+          <listitem>
+            <para><emphasis>aud</emphasis>: Audience, a list of strings representing the recipients that the JWT is intended for. Each principal intended to process the JWT shall identify itself with a value in the audience claim. If the principal processing the claim does not identify itself with a value in the audience claim when this claim is present, then the JWT shall be rejected.</para>
+          </listitem>
+          <listitem>
+            <para><emphasis>exp</emphasis>: Expiration Time.</para>
+          </listitem>
+          <listitem>
+            <para><emphasis>nbf</emphasis>: Not before.</para>
+          </listitem>
+        </itemizedlist>
+      </para>
+      <para>For the <emphasis>exp</emphasis>, <emphasis>nbf</emphasis> claims, a device shall reject a token when the current time is not within the range of claims nbf and exp . It shall reject a token when at least one of the claims is missing.</para>
+      <para>The JWT shall include the <emphasis>roles</emphasis> claim, as defined within [RFC 7643]:</para>
```

Comment:
Thinking about this I think it will be very hard to get identity providers to include this in their tokens. I think we need a more flexible approach. I was thinking of some kind of claim mapping, so you can map Admin/Operator/Viewer to the claims that the provider is specifying.

Reply:
Can you please explain a little bit more your idea?

Reply:
I'll try to make a proposal for the Singapore meeting

Reply:
Also, based on the following https://leastprivilege.com/2016/12/16/identity-vs-permissions/, we should try to separate authentication (identity) from authorization (permissions/roles)

Reply:
@jflevesque-genetec I do not understand you last comment. What shoudl we change in the specs?

For the ONVIF prefix. I think it is a good idea.

Reply:
Essentially, the blogpost mentions that with experience, using roles or authorization from the token limits you a lot, because your permissions need to be 100% static for the whole duration of your session. If for any reason they should change with time, having them in the token is a bad idea. Since we are currently mapping the user level in the token, that means that if I would change the user level of that user in my cloud system, he/she/they would still have their old user level until their token expires and they request a new one, which is suboptimal.

Thinking about it more, I am not certain it is something we can make work because the camera has to rely on the token to determine the access class of the user, so maybe we can't remove the roles. But having the prefix would be the minimum requirement for roles I think, otherwise you cannot have a multipurpose token and you are forcing every client cloud infrastructure to have an id provider specific for ONVIF devices

Reply:
I kind of agree. Axis is currently doing an analysis of how we should structure tokens for our own services and have looked at how other organizations have done it, like github and AWS, but we have not completed that work.

Reply:
but in this case, regenerating a token with a different role wouldn't be enough?

I mean, it's like when you add a user to a group in unix. To make this effective, you have to logout and repeat login to make sure it works.